tests/test_caching.py

```diff
@@ -278,6 +279,28 @@ def test_environment_variable_instantiation(rig: CloudProviderTestRig, tmpdir):
         os.environ["CLOUPATHLIB_FILE_CACHE_MODE"] = original_env_setting
 
 
+def test_environment_variable_local_cache_dir(rig: CloudProviderTestRig, tmpdir):
```

Comment:
`tmpdir` is deprecated and we should use `tmp_path` instead: https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

Reply:
Added issue #370 to make this change everywhere